src/gui/mzroll/isotopeswidget.cpp

```diff
@@ -166,9 +166,19 @@ void IsotopeWidget::computeIsotopes(string f) {
 
 		mzLink link;
 		mzSample* sample = isotopeParameters->_scan->getSample();
+		bool filterIsotope = false;
 		
-		if (isotopeDetector->filterIsotope(x, isotopePeakIntensity, parentPeakIntensity, sample))
-			continue;
+		if (isotopeParameters->_group)
+		{
+			filterIsotope = isotopeDetector->filterIsotope(x, isotopePeakIntensity, parentPeakIntensity, sample, isotopeParameters->_group);
+		}
+		else
+		{
+			filterIsotope = isotopeDetector->filterIsotope(x, isotopePeakIntensity, parentPeakIntensity, sample);
+		}
```

Comment:
this can be converted to one line of code. 
just use it as this:
`filterIsotope = isotopeDetector->filterIsotope(x, isotopePeakIntensity, parentPeakIntensity, sample, isotopeParameters->_group);}`

and make sure that isotopeParameters->_group: 
-  is initialized to null in `IsotopeLogic` 
-  is only used in `filterIsotope` when it's not null

Reply:
My bad. Missed this somehow. I've pushed the requested changes; you can review this again.